ml_garden/core/pipeline.py

```diff
@@ -32,10 +32,13 @@ class Pipeline:
         "explainer",
     ]
 
+    SUPPORTED_TASKS = {"classification", "regression"}
```

Comment:
I think it's a good moment to make this a `str, Enum`. The key advantage is that it allows good type-hints (as `task: SupportedTasks` is more descriptive than `task: str`). If this is done, I'd declare the enum quite top-level and import it everywhere where it's needed.

At the very least, I'd use `Literal` in the type hints instead of bare `str`, though that necessitates more maintenance.